### DISCUSSION TITLE
An error occurred when using the Modbus Service to connect to a real sensor！

### DISCUSSION THREAD

**Discussion by HuangYuFe** [edgexfoundry/community#298] [Q&A]
I tried to use the EdgeX project to complete the integration of real sensor devices based on the ModbusRTU protocol. I completed the deployment of EdgeX using Kuberntes' Deployment deployer。
![屏幕截图 2024-08-05 173522](https://github.com/user-attachments/assets/385962d3-8f8e-420f-94c8-f1c45abc4d40)
In addition, I have completed the access of real sensor devices based on ModbusTCP, which can complete the data collection. 
![屏幕截图 2024-08-05 172918](https://github.com/user-attachments/assets/eec2d49d-16cd-491b-a32c-8fc43de45e27)
But when I try to collect ModbusRTU device data via the USB hardware interface, I get a problem.
![屏幕截图 2024-08-05 172951](https://github.com/user-attachments/assets/4dab35a0-cddf-4a28-b915-d4a8eefd7ee3)
I checked the deployment information of the EdgeX application according to the permission denial prompted by the log message, and I confirmed that the mount path is /dev/ttyUSB0 and I have the r/w permission
![屏幕截图 2024-08-05 173215](https://github.com/user-attachments/assets/43a24325-5337-4455-b6c4-295b0a54e9d3)
So, it's not clear to me what the cause of the error is, here's the YAML content of my Kuberntes deployer:
![屏幕截图 2024-08-05 173417](https://github.com/user-attachments/assets/e71a519e-88b6-4db4-a870-0e6ddf4c9fd4)

**Comment by cloudxxx8** [comment]
it's the deployment and permission setting issue.  we only have this reference for suggestion
https://docs.edgexfoundry.org/3.1/microservices/device/services/device-modbus/GettingStarted/#set-up-the-modbus-rtu-device

**Comment by HuangYuFe** [comment]
@cloudxxx8 
Okay, but I'm just referring to that link for deployment. The difference is that I'm using Kubernetes instead of Docker Compose to deploy the EdgeX.
